core/bin/zksync_tee_prover/src/api_client.rs

```diff
@@ -0,0 +1,146 @@
+use reqwest::Client;
+use secp256k1::{ecdsa::Signature, PublicKey, Secp256k1, SecretKey};
+use serde::{de::DeserializeOwned, Serialize};
+use url::Url;
+use zksync_basic_types::H256;
+use zksync_prover_interface::{
+    api::{
+        RegisterTeeAttestationRequest, RegisterTeeAttestationResponse, SubmitProofResponse,
+        SubmitTeeProofRequest, TeeProofGenerationDataRequest, TeeProofGenerationDataResponse,
+    },
+    inputs::TeeVerifierInput,
+    outputs::L1BatchTeeProofForL1,
+};
+use zksync_types::{tee_types::TeeType, L1BatchNumber};
+
+/// Configuration for the [`zksync_proof_data_handler::run_server`] API client that serves the TEE
+/// prover by:
+/// 1. Providing proof generation data.
+/// 2. Submitting proofs.
+/// 3. Submitting attestation quotes.
+#[derive(Debug)]
+pub(crate) struct ApiClient {
+    /// Base URL for the [`zksync_proof_data_handler`] API. Optionally, include the port number,
+    /// e.g., `https://127.0.0.1:3320`.
+    api_base_url: Url,
+    /// HTTP client used to send requests to the API.
+    http_client: Client,
+}
+
+impl ApiClient {
+    pub fn new(api_base_url: Url, http_client: Client) -> Self {
+        ApiClient {
+            api_base_url,
+            http_client,
+        }
+    }
+
+    async fn send_http_request<Req, Resp>(
+        &self,
+        request: Req,
+        endpoint: Url,
+    ) -> Result<Resp, reqwest::Error>
+    where
+        Req: Serialize,
+        Resp: DeserializeOwned,
+    {
+        tracing::info!("Sending request to {}", endpoint);
+
+        self.http_client
+            .post(endpoint)
+            .json(&request)
+            .send()
+            .await?
+            .error_for_status()?
+            .json::<Resp>()
+            .await
+    }
+
+    pub async fn register_attestation(
+        &self,
+        attestation_quote_bytes: Vec<u8>,
+        signing_key: &SecretKey,
```

Comment:
You pass sensitive information (secret key) just to construct public information (public key). It's better to construct it outside and pass. Rule of thumb: the components should have the least amount of information possible to work correctly.